src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,40 +1,63 @@
 import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { FILTER_ACTIONS, FILTER_SELECTORS } from '../../features/filter';
+import { Status } from '../../types/Status';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useDispatch();
+
+  const query = useSelector(FILTER_SELECTORS.getQuery);
+  const status = useSelector(FILTER_SELECTORS.getStatus);
+
+  const setQuery = (value: string) => {
+    dispatch(FILTER_ACTIONS.setQuery(value));
+  };
+
+  const setStatus = (chosenStatus: Status) => {
+    dispatch(FILTER_ACTIONS.setStatus(chosenStatus));
+  };
+
   return (
     <form
       className="field has-addons"
       onSubmit={event => event.preventDefault()}
     >
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
+          <select
+            data-cy="statusSelect"
+            value={status}
+            onChange={(event) => setStatus(event.target.value as Status)}
+          >
             <option value="all">All</option>
             <option value="active">Active</option>
             <option value="completed">Completed</option>
           </select>
         </span>
       </p>
-
       <p className="control is-expanded has-icons-left has-icons-right">
         <input
           data-cy="searchInput"
           type="text"
           className="input"
           placeholder="Search..."
+          value={query}
+          onChange={(event) => setQuery(event.target.value)}
         />
         <span className="icon is-left">
           <i className="fas fa-magnifying-glass" />
         </span>
-
-        <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button
-            data-cy="clearSearchButton"
-            type="button"
-            className="delete"
-          />
-        </span>
+        {query.trim().length !== 0 && (
+          <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
```

Comment:
Try to fix eslint problems, not just disable it
https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/label-has-associated-control.md